openpathsampling/visualize.py

```diff
@@ -1,502 +1,394 @@
+        return group
 
-        self.start_x = self.margin
-        self.start_y = self.margin
+    def vertical_label(self, x, y, text, cls=None):
+        if cls is None:
+            cls = list()
 
-        self.document = svgwrite.Drawing()
-        self.document['width'] = str(self._width()) + 'px'
-        self.document['height'] = str(self._height()) + 'px'
+        cls += ['v-label']
 
-        for obj in self.obj:
-            parts = obj()
-            map(self.document.add, parts)
+        group = self.g(
+            class_=self.c(cls)
+        )
 
-        # self.document.add(self.document.rect(
-        #            insert = (0, 0),
-        #            size = (self._width(), self._height()),
-        #            fill = 'none',
-        #            stroke = 'black'
-        #        ))
+        group.translate(x, y)
 
-        return self.document
+        group.add(
+            self.text(
+                text=str(text),
+                insert=(0, 0),
+            )
+        )
 
-    def to_svg(self):
-        doc = self.to_document()
-        return doc.tostring()
+        return group
 
-    def to_html(self, svg=None):
-        if svg is None:
-            svg = self.to_svg()
+    def rectangle(self, x, y, w, h, cls=None):
+        if cls is None:
+            cls = list()
 
+        return self.rect(
+            class_=self.c(cls),
+            insert=self.xy(x, y),
+            size=self.wh(w, h),
+        )
+
+    def to_svg(self):
+        return self.tostring()
+
+    def to_html(self):
+        svg_source = self.to_svg()
         html = '<!DOCTYPE html><html style="margin:0px; padding:0px; width:100%;">' + \
-               svg + '<body style="margin:0px; padding:0px;"></body></html>'
+               svg_source + '<body style="margin:0px; padding:0px;"></body></html>'
 
         return html
 
     def _height(self):
-        return self._h(self.height) + self.margin * 2
+        return self.h(self.height) + self.margin * 2
 
     def _width(self):
-        return self._w(self.width) + self.margin * 2
+        return self.w(self.width) + self.margin * 2
 
     def write_html(self, file_name='tree.html'):
         with open(file_name, 'w') as f:
             f.write(self.to_html())
 
     def save_pdf(self, file_name='tree.pdf'):
-        h = self._height()
-        w = self._width()
-
-        h_margin = 3.5
-        v_margin = 3.5
-
-        page_height = str(25.4 / 75.0 * h + v_margin) + 'mm'
-        page_width = str(25.4 / 75.0 * w + h_margin) + 'mm'
-
         with open('tree_xxx.html', 'w') as f:
             f.write(self.to_html())
 
-        bash_command = "wkhtmltopdf -l --page-width " + page_width + \
-                      " --page-height " + page_height + \
-                      " --disable-smart-shrinking " + \
-                      "-B 1mm -L 1mm -R 1mm -T 1mm tree_xxx.html " + file_name
+        bash_command = "open tree_xxx.html " + file_name
 
         os.system(bash_command)
```

Comment:
Can we find a way to do this that doesn't rely on a bash command? I believe that `open` is Mac-specific, and the previous `wkhtmltopdf` doesn't exist on my system. In general, I'd suggest avoiding the approach of writing a temporary html file (which you don't remove?!) and using an external process to convert from that. It's hard to make it platform-independent.

If we can't write to PDF directly, then we just give the options of writing to SVG or HTML. If the user wants to convert that to PDF, they can google "convert HTML to PDF" and find a solution for their system.